gnucash/import-export/import-main-matcher.c

```diff
@@ -117,10 +121,22 @@ static gboolean gnc_gen_trans_onButtonPressed_cb (GtkTreeView *treeview,
 static gboolean gnc_gen_trans_onPopupMenu_cb (GtkTreeView *treeview,
                                               GdkEvent *event,
                                               GNCImportMainMatcher *info);
+gboolean view_selection_function (GtkTreeSelection *selection,
+                         GtkTreeModel *model,
+                         GtkTreePath *path,
+                         gboolean path_currently_selected,
+                         gpointer data);
 static void refresh_model_row (GNCImportMainMatcher *gui, GtkTreeModel *model,
                                GtkTreeIter *iter, GNCImportTransInfo *info);
 /* end local prototypes */
 
+/* define the add_toggled gboolean flag for use by the vie-selection_funcc-tion
+   to avoid a row being selected immediately after the row's Add checkbox has 
+   been toggled on. This casues a problem if the U+R or R checkboxes are toggled
+   without the row being unselected
+*/
+gboolean add_toggled = FALSE;
```

Comment:
I probably would not do this, instead I would add the gboolean to _main_matcher_info with a suitable comment and then you will need to add the info structure to the function setup and access it from there...

    gtk_tree_selection_set_select_function
                               (selection,
                                view_selection_function,
                                info,
                                NULL);

Reply:
OK hadn't thought of that. I'll give that a go. I was not happy about putting a global in.

Reply:
Will do Bob

Reply:
Bob Worked like a charm